bin/dev/create_helm_charts_pr.sh

```diff
@@ -0,0 +1,71 @@
+#!/bin/bash
+
+# This script is used to open a PR against this repository:
+# https://github.com/suse/kubernetes-charts-suse-com
+# It will fetch the bundle specified by CAP_BUNDLE and will update the helm
+# charts in the above repo.
+
+set -e
+
+if [ -z $GITHUB_USER  ]; then
+  echo "GITHUB_USER environment variable not set"
+  exit 1
+fi
+
+if [ -z $GITHUB_PASSWORD ]; then
+  echo "GITHUB_PASSWORD environment variable not set"
+  exit 1
+fi
+
+if [[ -z "${CAP_BUNDLE}" ]]; then
+  echo "CAP_BUNDLE not set"
+  exit 1
+fi
+
+TMP=$(mktemp -d -p $PWD)
+#trap "rm -rf ${TMP}" EXIT
+
+pushd $TMP
+
+# Get the "hub" cli
+# https://hub.github.com/hub.1.html
+wget https://github.com/github/hub/releases/download/v2.3.0-pre10/hub-linux-amd64-2.3.0-pre10.tgz
+tar xvzf hub-linux-*.tgz
+cp hub-linux-*/bin/hub hub
+HUB=$(readlink -f hub)
+chmod +x $HUB
+
+# escape user and password
+USER=$(ruby -e "require 'cgi'; puts CGI.escape(ENV['GITHUB_USER'])")
+PASS=$(ruby -e "require 'cgi'; puts CGI.escape(ENV['GITHUB_PASSWORD'])")
+# Clone the kubernetes-charts-suse-com github repo
+$HUB clone https://${USER}:${PASS}@github.com/SUSE/kubernetes-charts-suse-com.git
```

Comment:
It looks like `hub` should be handling auth via the environment variables already; do we need to put them in the URL too?

(Using ruby means we'll need to make sure we _have_ ruby on the slaves… which we actually do, but still.)

Reply:
`hub` only uses the `GITHUB_USER` and `GITHUB_PASSWORD` environment variables to authenticate against the github API. The prompt you get when using `hub clone` with `https` comes from `git` itself. `git` doesn't use env vars to authenticate. There are other ways to do that, but in the end [it seems](https://git-scm.com/docs/git-credential-store#_storage_format) that they might still need escaping because all they store is this url above.

Reply:
Actually, when is this script run? Via Jenkins or via somebody typing it on the command line?  Because we could totally just use the Jenkins-stored creds for that…

Reply:
It will run on Jenkins. These are the Jenkins creds coming as GITHUB_USER and GITHUB_PASSWORD env vars. Maybe we could have Jenkins pulling in the repo (can we?) but we would still need the creds in order to push later in this script.

Reply:
Yeah, we can have Jenkins pull in the repo (that's how we check out the Jenkinsfile for the main scf pipelines).

I guess the other option is to use `GIT_ASKPASS` (but that needs to be a path to a program that echos the password, not the password itself), and it's not necessarily any better.

Reply:
We completely removed this escaping logic by setting the `GIT_SSH_COMMAND` in the wrapper script in Jenkins. `git clone` now works with ssh.